src/DGmethods/DGModels.jl

```diff
@@ -0,0 +1,48 @@
+module DGModels
+export DGModel, @model
+
+abstract type DGModel{Dim, N} end
+
+macro model(expr)
+    @assert expr.head === :<:
+    def = Expr(:struct, false, expr, Expr(:block))
+    methods = (
+        :flux!,
+        :numerical_flux!,
+        :numerical_boundary_flux!,
+        :gradient_transform!,
+        :viscous_transform!,
+        :viscous_penalty!,
+        :viscous_boundary_penalty!,
+        :source!,
+        :hasbctype,
+        :nstate,
+        :nviscstate,
+        :ngradstate,
+        :nauxstate,
+        :states_grad)
+    imports = (Expr(:., m) for m in methods)
+
+    Expr(:toplevel,
+        Expr(:import, Expr(Symbol(":"), Expr(:., fullname(@__MODULE__)...), imports...)),
```

Comment:
I think we should keep imports out of the macro for now. We can add them in later if need be.